File: c3-platform/c3-search/src/main/scala/org/aphreet/c3/platform/search/ext/FieldWeights.java
package org.aphreet.c3.platform.search.ext;

import java.util.Collections;
import java.util.Map;

/**
 * Created by IntelliJ IDEA.
 * User: Ildar
 * Date: 02.02.2011
 * Time: 0:37:34
 */
public class FieldWeights {

    private Map<String, Float> weights;

    public FieldWeights(Map<String, Float> weights) {
        this.weights = Collections.unmodifiableMap(weights);
    }

    public boolean containsField(String key) {
        return weights.containsKey(key);
    }

    public float getBoostFactor(String key, float defaultValue) {

        Float value = weights.get(key);

        if(value != null){
            return value;
        }else{
            return defaultValue;
        }
    }

    public String[] getFields() {
        return weights.keySet().toArray(new String[weights.size()]);
    }

}

File: c3-platform/c3-search-lucene/src/main/scala/org/aphreet/c3/platform/search/lucene/impl/FieldWeights.scala
package org.aphreet.c3.platform.search.lucene.impl

import collection.Map
import java.lang.Float

class FieldWeights(val weights: Map[String, Float]) {

  def containsField(key: String): Boolean = {
    weights.contains(key)
  }

  def getBoostFactor(key: String, defaultValue: Float): Float = {
    weights.getOrElse(key, defaultValue)
  }

  def getFields: Array[String] = {
    weights.keys.toArray
  }
}
